### PR TITLE
Migrate to Slate 0.25

### PR DESCRIPTION
**Pull request by erquhart**
Migrate from Slate 0.21 to 0.25 - related changelog [here](https://github.com/ianstormtaylor/slate/blob/master/packages/slate/Changelog.md).

### CODE REVIEW

**Comment by tech4him1** [comment]
@erquhart This is weird -- it looks like the commits are out of order (upgrade to 0.25 comes before 0.22), but when I look at each diff, they are valid, but not in order?

File: package.json [hunk 1]
```diff
@@ -150,10 +156,12 @@
     "remark-rehype": "^2.0.0",
     "remark-stringify": "^3.0.1",
     "semaphore": "^1.0.5",
-    "slate": "^0.21.0",
-    "slate-edit-list": "^0.7.1",
-    "slate-edit-table": "^0.10.1",
-    "slate-soft-break": "^0.3.0",
+    "slate": "^0.25.0",
+    "slate-edit-list": "^0.8.0",
```
Comment (tech4him1) [inline package.json]: `warning "slate-edit-list@0.8.0" has incorrect peer dependency "slate@^0.23.0".`

Comment (erquhart) [reply package.json]: Yeah, I think that's just where `slate-edit-list` is at, it's the latest version.

Comment (tech4him1) [reply package.json]: They do use `Block.create()` here: https://github.com/GitbookIO/slate-edit-list/tree/master/lib/changes, in `decreaseItemDepth.js` and `increaseItemDepth.js`, I don't know if that could present a problem at all.

Comment (tech4him1) [reply package.json]: https://github.com/GitbookIO/slate-edit-list/blob/2ab5c0a581c71890f351069658580e620d72686f/lib/changes/decreaseItemDepth.js#L41-L45
https://github.com/GitbookIO/slate-edit-list/blob/2ab5c0a581c71890f351069658580e620d72686f/lib/changes/increaseItemDepth.js#L57-L61

File: src/components/Widgets/Markdown/MarkdownControl/VisualEditor/__tests__/__snapshots__/parser.spec.js.snap [hunk 2]
```diff
@@ -10,7 +10,11 @@ Object {
         Object {
           "data": undefined,
           "kind": "text",
-          "text": "H1",
+          "ranges": Array [
```
Comment (tech4him1) [inline src/components/Widgets/Markdown/MarkdownControl/VisualEditor/__tests__/__snapshots__/parser.spec.js.snap]: I didn't see in the changelog exactly why this was needed, could you explain?

Comment (erquhart) [reply src/components/Widgets/Markdown/MarkdownControl/VisualEditor/__tests__/__snapshots__/parser.spec.js.snap]: It's actually not in the changelog, but Slate text nodes used to come in two flavors, either they'd have a simple "text" property with the value, or else a "ranges" property having an array of objects representing text with associated marks. They quietly switched to using "ranges" only.

File: package.json (hunk 1 above)
Comment (erquhart) [reply package.json]: They're manually inserting chlidren, so those created blocks will never have no children. Not an issue. Thanks for checking their source!

**Pull request merged**